commit 2e30d561ae96252b90c6900d932cf1dd2f8df012
Author: Radomir Dopieralski <openstack@sheep.art.pl>
Date:   Wed Feb 7 19:32:42 2018 +0100

    Fix collectstatic in case of theme customization
    
    When doing theme customization with templates
    collectstatic failes with a traceback
    
    Change-Id: I0fdb80ddde1f73b657d445c75b55636d6f2e8d88
    Closes-bug: #1744239
    (cherry picked from commit dc0bce63a03e37e9f3f181930ba6d6b7c8b1346e)

diff --git a/openstack_dashboard/utils/settings.py b/openstack_dashboard/utils/settings.py
index 497c19f14..6326cddbb 100644
--- a/openstack_dashboard/utils/settings.py
+++ b/openstack_dashboard/utils/settings.py
@@ -15,6 +15,7 @@ from importlib import import_module
 import logging
 import os
 import pkgutil
+import re
 
 from horizon.utils import file_discovery
 from openstack_dashboard import theme_settings
@@ -327,9 +328,10 @@ def find_static_files(
             #       'framework/widgets/help-panel/help-panel.html'
             #   ] = 'themes/material/templates/framework/widgets/\
             #        help-panel/help-panel.html'
-            (templates_part, override_path) = theme_file.split('/templates/')
-            template_overrides[override_path] = 'themes/' + \
-                                                theme_name + theme_file
+            override_path = re.sub(r'^(|.*/)templates/', '', theme_file)
+            template_overrides[override_path] = os.path.join('themes',
+                                                             theme_name,
+                                                             theme_file)
 
         discovered_files['template_overrides'] = template_overrides
 
